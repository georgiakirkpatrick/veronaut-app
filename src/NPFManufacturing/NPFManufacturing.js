import React from 'react'
import FormPage from '../FormPage/FormPage'
import FormButton from '../FormButton/FormButton'
import FormCheckboxSection from '../FormCheckboxSection/FormCheckboxSection'
import FormDropdown from '../FormDropdown/FormDropdown'
import FormFieldset from '../FormFieldset/FormFieldset'
import NPFFooter from '../NPFFooter/NPFFooter'
import FormTextInput from '../FormTextInput/FormTextInput'
import FormUrlInput from '../FormUrlInput/FormUrlInput'
import FormPopUp from '../FormPopUp/FormPopUp'
import FormPromptWithSub from '../FormPromptWithSub/FormPromptWithSub'
import config from '../config'

const NPFManufacturing = props => {
    const makeCountryOptions = () => {
        const countries = props.fabricProps.countries.map((country, index) => ({
            id: index + 2,
            text: country.text,
            value: index + 2
        }))

        return [
            {
                id: 0,
                text: 'Select a country',
                value: 0
            },
            ...countries
        ]
    }

    const makeFactoryOptions = () => {
        const factoryQty = props.fabricProps.factoryList.length
        const mills = props.fabricProps.factoryList.slice(1, factoryQty)
        const alphaMills = mills.sort((a, b) => a.english_name > b.english_name ? 1 : -1)

        const formatedMills = alphaMills.map(mill => (
            {
                id: mill.id,
                text: mill.english_name,
                value: mill.id
            }
        ))

        return [
            {
                id: 0,
                text: `Select a factory`,
                value: 0
            },
            {
                id: 1,
                text: 'Not disclosed',
                value: 1
            },
            ...formatedMills
        ]
    }

    const addFactory = (stage, factory) => {
        props.fabricProps.setFactoryList([
            ...props.fabricProps.factoryList,
            {
                "id": Number(factory.id),
                "english_name": factory.english_name,
                "country": Number(factory.country),
                "website": factory.website,
                "notes": factory.notes,
                "approved_by_admin": factory.approved_by_admin
            }
        ])
            
        const newFactory = {...props.sewFact}
        newFactory.factoryId = factory.id

        if (stage === 's') {
            props.setSewFact(newFactory)
        } else if (stage === 'c') {
            props.setCutFact(newFactory)
        }
    }

    const submitNewFactory = stage => {
        const data = {
            "english_name": props.fabricProps.newFact.name,
            "country": props.fabricProps.newFact.countryId,
            "website": props.fabricProps.newFact.website,
            "notes": props.fabricProps.newFact.notes,
            "approved_by_admin": false
        }

        const postRequestParams = {
            method: 'POST',
            headers: { 'Content-type': 'application/json' },
            body: JSON.stringify(data)
        }

        fetch(`${config.API_URL}/api/factories`, postRequestParams)
            .then(response => {
                if (response.status >= 400) {
                    throw new Error("Server responded with an error!")
                }
                return response.json()
            })
            .then(responseJson => {
                addFactory(stage, responseJson)
            })

        props.fabricProps.setNewFact({
            name: '',
            countryId: 0,
            website: '',
            notes: ''
        })
    }

    const handleFactoryPopSubmit = stage => {
        const missingFields = []

        Object.keys(props.fabricProps.newFact).forEach(key => {
            if (props.fabricProps.newFact[key] === '' || 0) {
                missingFields.push(key.replace( /([A-Z])/g, " $1" ).toLowerCase())
            }
        })

        if (missingFields.length === 1) {
            alert(`Please complete the '${missingFields[0]}' field`)
        } else if (missingFields.length > 1) {
            alert(`Please complete the following fields: ${missingFields.map(field => `
                ${field}`)}
            `)
        } else if (missingFields.length === 0) {
            submitNewFactory(stage)
            handlePopUpClose()

        }
    }

    // NEW FACTORY
    const sewFactPopUpStatus = () => {
        if (props.sewFactPopUp === true) {
            return 'FormPopUp__pop-up active'
        }
        return 'FormPopUp__pop-up'
    }

    const cutFactPopUpStatus = () => {
        if (props.cutFactPopUp === true) {
            return 'FormPopUp__pop-up active'
        }
        return 'FormPopUp__pop-up'
    }

    const newFactChangeInput = event => {
        const newFactFields = {...props.fabricProps.newFact}
        newFactFields[event.target.name] = event.target.value
        props.fabricProps.setNewFact(newFactFields)
    }

    const handlePopUpClose = () => {
        props.fabricProps.setCertPopUp(false)
        props.setSewFactPopUp(false)
        props.setCutFactPopUp(false)

    }

    // SEWING FACTORY
    const sewingChangeInput = event => {
        const sewingFactory = {...props.sewFact}
        sewingFactory[event.target.name] = event.target.value
        props.setSewFact(sewingFactory)
    }

    // CUTTING FACTORY
    const cuttingChangeInput = event => {
        const cuttingFactory = {...props.cutFact}
        cuttingFactory[event.target.name] = event.target.value
        props.setCutFact(cuttingFactory)
    }

    // CERTIFICATIONS
    const addCertification = certification => {
        const newCertList = [
            ...props.fabricProps.certificationList,
            {
                id: certification.id,
                name: certification.name,
                text: certification.english_name,
                website: certification.website,
                approved: certification.approved_by_admin
            }
        ]
            
        newCertList.sort((a, b) => a.text > b.text ? 1 : -1)
        props.fabricProps.setCertificationList(newCertList)
    }

    const certificationChange = event => {
        props.setCertChecks({
            ...props.certChecks, 
            [event.target.name]: !props.certChecks[event.target.name]})
    }

    const certPopUpStatus = () => {
        if (props.fabricProps.certPopUp === true) {
            return 'FormPopUp__pop-up active'
        }
        return 'FormPopUp__pop-up'
    }
    
    const cmtNotesChange = event => {
        const newNotes = event.target.value
        props.setCmtNotes(newNotes)
    }

    const newCertChangeInput = event => {
        const newCertFields = {...props.fabricProps.newCert}
        newCertFields[event.target.name] = event.target.value
        props.fabricProps.setNewCert(newCertFields)
    }

    const submitNewCert = () => {
        const data = {
            "english_name": props.fabricProps.formatName(props.fabricProps.newCert.name),
            "website": props.fabricProps.formatUrl(props.fabricProps.newCert.website),
            "approved_by_admin": false
        }

        const postRequestParams = {
            method: 'POST',
            headers: { 'Content-type': 'application/json' },
            body: JSON.stringify(data)
        }

        fetch(`${config.API_URL}/api/certifications`,
            postRequestParams
        )
        .then(response => {
            if (response.status >= 400) {
                throw new Error('Server responded with an error!')
            }
            return response.json()
        })
        .then(responseJson => {
            addCertification(responseJson)
        })
    }

    const handleCertPopSubmit = () => {
        const missingFields = []

        Object.keys(props.fabricProps.newCert).forEach(key => {
            props.fabricProps.newCert[key] === '' && missingFields.push(
                key.replace( /([A-Z])/g, " $1" ).toLowerCase()
            )
        })

        if (missingFields.length === 1) {
            alert(`Please complete the '${missingFields[0]}' field`)
        } else if (missingFields.length > 1) {
            alert(`Please complete the following fields: ${missingFields.map(field => `
                ${field}`)}
            `)
        } else if (missingFields.length === 0) {
            submitNewCert()
            handlePopUpClose()
            props.fabricProps.setNewCert(
                {
                    name: '',
                    website: ''
                }
            )
        }
    }

    // const nextButton = () => {props.fabricProps.setPage(props.fabricProps.currentPage + 1)}

    const nextButton = () => {
        const missingFields = []
        
        const requiredSewFields = {
            'countryId': 'sewing location',
            'factoryId': 'sewing factory'
        }

        Object.keys(requiredSewFields).forEach(key => {
            if (props.sewFact[key] === 0) {
                missingFields.push(requiredSewFields[key])
            }
        })

        const requiredCutFields = {
            'countryId': 'cutting location',
            'factoryId': 'cutting factory'
        }

        Object.keys(requiredCutFields).forEach(key => {
            if (props.cutFact[key] === 0) {
                missingFields.push(requiredCutFields[key])
            }
        })

        if (missingFields.length === 1) {
            alert(`Please complete the '${missingFields[0]}' field`)
        } else if (missingFields.length > 1) {
            alert(`Please complete the following fields: ${missingFields.map(field => `
                ${field}`)}
            `)
        } else if (missingFields.length === 0) {
            props.fabricProps.setPage(props.fabricProps.currentPage + 1)
        }
    }

    return (
        <div id='manufacturing'>
            <FormPage title='Manufacturing'>
                <FormPromptWithSub 
                    prompt='Enter the manufacturing information provided on the product page.'
                    promptSubtitle='If the product page does not provide factory names or locations, select "not disclosed" from the dropdown.'
                />

                <FormFieldset
                    prompt='Sewing factory'
                >
                    <FormDropdown
                        id='sewing-location'
                        name='countryId'
                        prompt='Location'
                        currentValue={props.sewFact.countryId}
                        options={makeCountryOptions()}
                        handleChange={event => sewingChangeInput(event)} 
                    />

                    <FormDropdown
                        id='sewing-factory'
                        name='factoryId'
                        prompt='Factory'
                        currentValue={props.sewFact.factoryId}
                        options={makeFactoryOptions()}
                        handleChange={event => sewingChangeInput(event)} 
                    />

                    <FormButton 
                        buttonText='ADD A FACTORY'
                        handleClick={() => props.setSewFactPopUp(true)} 
                    />
                </FormFieldset>
                
                <FormFieldset 
                    prompt='Cutting factory'
                    subprompt='If the product page does not provide a factory location, select "Not disclosed"'
                >
                    <FormDropdown
                        id='cutting-location'
                        name='countryId'
                        prompt='Location'
                        currentValue={props.cutFact.countryId}
                        options={makeCountryOptions()}
                        handleChange={event => cuttingChangeInput(event)} 
                    />
                    <FormDropdown
                        id='cutting-factory'
                        name='factoryId'
                        prompt='Factory'
                        currentValue={props.cutFact.factoryId}
                        options={makeFactoryOptions()}
                        handleChange={event => cuttingChangeInput(event)} 
                    />

                    <FormButton
                        buttonText='ADD A FACTORY'
                        handleClick={() => props.setCutFactPopUp(true)}
                    />
                </FormFieldset>

                <FormFieldset
                    prompt='Does the product have any of the following certifications?'
                    subprompt='Check all that apply'
                >
                    <FormCheckboxSection
                        prompt=''
                        options={props.fabricProps.certificationList}
                        selectedOptions={props.certChecks}
                        handleChange={event => certificationChange(event)}
                    />
                    <FormButton 
                        buttonText='ADD A CERTIFICATION'
                        handleClick={() => props.fabricProps.setCertPopUp(true)} 
                    />
                </FormFieldset>

                <FormFieldset
                    prompt='Are there any additional notes about how the product was cut, sewn, or finished?'
                    subprompt=''
                >
                    <FormTextInput
                        id='additional-notes'
                        name='cmtNotes'
                        currentValue={props.cmtNotes}
                        handleChange={event => cmtNotesChange(event)}
                    />
                </FormFieldset>
            </FormPage>


            <NPFFooter 
                buttons='prevNext' 
                previousButton={() => props.setPage(props.currentPage - 1)} 
                nextButton={event => nextButton(event)}
            />

            <FormPopUp
                id='addSewFact'
                status={sewFactPopUpStatus()}
                title='New Factory'
                close={() => handlePopUpClose()}
                submit={() => handleFactoryPopSubmit('s')}
                buttonText='SUBMIT FACTORY'
                buttonType='submit'
            >
                <FormPromptWithSub 
                    prompt='Add a factory'
                    promptSubtitle={`Include the factory website if it is provided on the product page.  Copy and paste all available information about the factory in the 'notes' field`}
                />

                <FormTextInput 
                    id='sew-factory-name'
                    name='name'
                    prompt='Factory name'
                    currentValue={props.fabricProps.newFact.name}
                    handleChange={event => newFactChangeInput(event)} 
                />

                <FormDropdown
                    id='sew-factory-location'
                    name='countryId'
                    prompt='Factory location'
                    currentValue={props.fabricProps.newFact.countryId}
                    options={makeCountryOptions()}
                    handleChange={event => newFactChangeInput(event)} 
                />

                <FormUrlInput 
                    id='sew-factory-website'
                    name='website'
                    prompt='Factory website'
                    currentValue={props.fabricProps.newFact.website}
                    handleChange={event => newFactChangeInput(event)}
                />

                <FormTextInput
                    id='sew-factory-notes'
                    name='notes'
                    prompt='Notes'
                    currentValue={props.fabricProps.newFact.notes}
                    handleChange={event => newFactChangeInput(event)}
                />
            </FormPopUp>

            <FormPopUp
                id='addCutFact'
                status={cutFactPopUpStatus()}
                title='New Factory'
                close={() => handlePopUpClose()}
                submit={() => handleFactoryPopSubmit('c')}
                buttonText='SUBMIT FACTORY'
                buttonType='submit'
            >
                <FormPromptWithSub 
                    prompt='Add a factory'
                    promptSubtitle={`Include the factory website if it is provided on the product page.  Copy and paste all available information about the factory in the 'notes' field`}
                />

                <FormTextInput 
                    id='cut-factory-name'
                    name='name'
                    prompt='Factory name'
                    currentValue={props.fabricProps.newFact.name}
                    handleChange={event => newFactChangeInput(event)} 
                />

                <FormDropdown
                    id='cut-factory-location'
                    name='countryId'
                    prompt='Factory location'
                    currentValue={props.fabricProps.newFact.countryId}
                    options={makeCountryOptions()}
                    handleChange={event => newFactChangeInput(event)} 
                />

                <FormUrlInput 
                    id='cut-factory-website'
                    name='website'
                    prompt='Factory website'
                    currentValue={props.fabricProps.newFact.website}
                    handleChange={event => newFactChangeInput(event)}
                />

                <FormTextInput
                    id='cut-factory-notes'
                    name='notes'
                    prompt='Notes'
                    currentValue={props.fabricProps.newFact.notes}
                    handleChange={event => newFactChangeInput(event)}
                />
            </FormPopUp>

            <FormPopUp
                id='cert-pop-up'
                status={certPopUpStatus()}
                title='New Certification'
                close={() => handlePopUpClose()}
                submit={() => handleCertPopSubmit()}
                buttonText='CREATE CERTIFICATION'
                buttonType='submit'
            >
                <FormPromptWithSub prompt='Add a certification' promptSubtitle='' />
                
                <FormTextInput 
                    id='man-cert-name'
                    name='name'
                    prompt='Certification name'
                    currentValue={props.fabricProps.newCert.name}
                    handleChange={event => newCertChangeInput(event)} 
                />
                
                <FormUrlInput
                    id='man-cert-website'
                    name='website'
                    prompt='Website'
                    currentValue={props.fabricProps.newCert.website}
                    handleChange={event => newCertChangeInput(event)}
                />
            </FormPopUp>
        </div>
    )  
}

NPFManufacturing.defaultProps = {
    certChecks: {},
    cmtNotes: '',
    cutFact: {
        countryId: '', 
        factoryId: ''
    },
    fabricProps: {
        certificationList: [],
        certPopUp: false,
        countries: [],
        currentPage: 0,
        factPopUp: false,
        factoryList: [],
        fiberPopUp: false,
        fiberTypeList: [],
        millPopUp: false,
        newCert: {
            name: '',
            website: ''
        },
        newFact: {
            name: '',
            countryId: '',
            website: '',
            notes: ''
        },
        newFiber: {
            name: ''
        },
        newMill: {
            name: '',
            countryId: '',
            website: '',
            notes: ''
        },
        setCertificationList: () => {},
        setFiberTypeList: () => {},
        setFactoryList: () => {},
        setCertPopUp: () => {},
        setFiberPopUp: () => {},

        setMillPopUp: () => {},
        setNewCert: () => {},
        setNewFact: () => {},
        setNewFiber: () => {},
        setNewMill: () => {},
        setPage: () => {},
        setSewFactPopUp: () => {},
        sewFactPopUp: false
    },
    setCertChecks: () => {},
    setCmtNotes:  () => {},
    setCutFact: () => {},
    setSewFact: () => {},
    sewFact: {
        countryId: '', 
        factoryId: ''
    }
}

export default NPFManufacturing