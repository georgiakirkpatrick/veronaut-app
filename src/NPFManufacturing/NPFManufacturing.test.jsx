import React from 'react'
import { render, screen } from '@testing-library/react'
import { describe, it, expect } from 'vitest'
import NPFManufacturing from './NPFManufacturing'

describe('NPFManufacturing', () => {
    it ('renders without crashing', () => {
        render(<NPFManufacturing />)
        screen.debug()
    })

    it ('render the UI as expected', () => {
        const tree = render(<NPFManufacturing />)
        expect(tree).toMatchSnapshot()
    })
})