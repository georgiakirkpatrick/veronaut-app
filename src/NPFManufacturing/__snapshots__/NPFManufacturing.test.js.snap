// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NPFManufacturing render the UI as expected 1`] = `
<div
  id="manufacturing"
>
  <section
    className="FormPage"
  >
    <h1>
      Manufacturing
    </h1>
    <form
      autoComplete="off"
      id="form-page"
    >
      <div
        className="FormPromptWithSub"
      >
        <p
          className="FormPromptWithSub__prompt"
        >
          Enter the manufacturing information provided on the product page.
        </p>
        <p
          className="FormPromptWithSub__subtitle"
        >
          If the product page does not provide factory names or locations, select "not disclosed" from the dropdown.
        </p>
      </div>
      <fieldset
        className="FormFieldset"
      >
        <h2>
          Sewing factory
        </h2>
        <div
          className="FormDropdown"
        >
          <label
            htmlFor="sewing-location"
          >
            Location
          </label>
          
          <select
            name="countryId"
            onChange={[Function]}
            required={true}
            value=""
          />
        </div>
        <div
          className="FormDropdown"
        >
          <label
            htmlFor="sewing-factory"
          >
            Factory
          </label>
          
          <select
            name="factoryId"
            onChange={[Function]}
            required={true}
            value=""
          >
            <option
              id={0}
              value={0}
            >
              Select a factory
            </option>
            <option
              id={1}
              value={1}
            >
              Not disclosed
            </option>
          </select>
        </div>
        <div
          className="FormPromptWithSub"
        >
          <p
            className="FormPromptWithSub__prompt"
          >
            Factory not listed above?
          </p>
          <p
            className="FormPromptWithSub__subtitle"
          >
            
          </p>
        </div>
        <button
          className="FormButton"
          onClick={[Function]}
          type="button"
        >
          Add a factory
        </button>
      </fieldset>
      <fieldset
        className="FormFieldset"
      >
        <h2>
          Cutting factory
        </h2>
        <div
          className="FormDropdown"
        >
          <label
            htmlFor="cutting-location"
          >
            Location
          </label>
          
          <select
            name="countryId"
            onChange={[Function]}
            required={true}
            value=""
          />
        </div>
        <div
          className="FormDropdown"
        >
          <label
            htmlFor="cutting-factory"
          >
            Factory
          </label>
          
          <select
            name="factoryId"
            onChange={[Function]}
            required={true}
            value=""
          >
            <option
              id={0}
              value={0}
            >
              Select a factory
            </option>
            <option
              id={1}
              value={1}
            >
              Not disclosed
            </option>
          </select>
        </div>
        <div
          className="FormPromptWithSub"
        >
          <p
            className="FormPromptWithSub__prompt"
          >
            Factory not listed above?
          </p>
          <p
            className="FormPromptWithSub__subtitle"
          >
            
          </p>
        </div>
        <button
          className="FormButton"
          onClick={[Function]}
          type="button"
        >
          Add a factory
        </button>
      </fieldset>
      <fieldset
        className="FormFieldset"
      >
        <h2>
          Does the product have any of the following certifications?
        </h2>
        <div
          className="FormCheckboxSection"
        >
          <div
            className="empty"
          />
          <div
            className="FormCheckboxSection__options-container"
          />
        </div>
        <div
          className="FormPromptWithSub"
        >
          <p
            className="FormPromptWithSub__prompt"
          >
            Certification not listed above?
          </p>
          <p
            className="FormPromptWithSub__subtitle"
          >
            
          </p>
        </div>
        <button
          className="FormButton"
          onClick={[Function]}
          type="button"
        >
          Add a certification
        </button>
      </fieldset>
      <fieldset
        className="FormFieldset"
      >
        <div />
        <div
          className="FormTextarea"
        >
          <label
            htmlFor="additional-notes"
          >
            Are there any additional notes about how the product was cut, sewn, or finished?
          </label>
          <textarea
            id="additional-notes"
            name="cmtNotes"
            onChange={[Function]}
            required={true}
            rows="1"
            value=""
          />
        </div>
      </fieldset>
    </form>
    <div>
      
    </div>
  </section>
  <footer
    className="NPFFooter"
  >
    <button
      className="NPFFooter__previous"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-left NPFFooter__left-arrow"
        data-icon="angle-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M41.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.3 256 246.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <span>
        PREVIOUS
      </span>
    </button>
    <button
      className="NPFFooter__next"
      onClick={[Function]}
      type="button"
    >
      <span>
        NEXT
      </span>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-right NPFFooter__right-arrow"
        data-icon="angle-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M278.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-160 160c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L210.7 256 73.4 118.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l160 160z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </footer>
  <section
    className="FormPopUp"
    id="addSewFact"
  >
    <button
      className="FormPopUp__close-button"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-xmark fa-lg "
        data-icon="xmark"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 384 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
    <header>
      <h1>
        New Factory
      </h1>
    </header>
    <div
      className="FormPromptWithSub"
    >
      <p
        className="FormPromptWithSub__prompt"
      >
        Add a factory
      </p>
      <p
        className="FormPromptWithSub__subtitle"
      >
        Include the factory website if it is provided on the product page.  Copy and paste all available information about the factory in the 'notes' field
      </p>
    </div>
    <div
      className="FormTextInput"
    >
      <label
        htmlFor="sew-factory-name"
      >
        Factory name
      </label>
      <input
        id="sew-factory-name"
        name="name"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="FormDropdown"
    >
      <label
        htmlFor="sew-factory-location"
      >
        Factory location.  Select 'Not disclosed' if not included on the product page.
      </label>
      
      <select
        name="countryId"
        onChange={[Function]}
        required={true}
        value=""
      />
    </div>
    <div
      className="FormUrlInput"
    >
      <label
        htmlFor="sew-factory-website"
      >
        Factory website.  Leave blank if not included on the product page.
      </label>
      <input
        id="sew-factory-website"
        name="website"
        onChange={[Function]}
        required={true}
        type="url"
        value=""
      />
    </div>
    <div
      className="FormTextarea"
    >
      <label
        htmlFor="sew-factory-notes"
      >
        Are there any additional notes about the sewing factory provided on the product page?  If so, copy them and paste them here.
      </label>
      <textarea
        id="sew-factory-notes"
        name="sewNotes"
        onChange={[Function]}
        required={true}
        rows="1"
        value=""
      />
    </div>
    <button
      className="FormButton"
      onClick={[Function]}
      type="submit"
    >
      SUBMIT FACTORY
    </button>
  </section>
  <section
    className="FormPopUp"
    id="addCutFact"
  >
    <button
      className="FormPopUp__close-button"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-xmark fa-lg "
        data-icon="xmark"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 384 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
    <header>
      <h1>
        New Factory
      </h1>
    </header>
    <div
      className="FormPromptWithSub"
    >
      <p
        className="FormPromptWithSub__prompt"
      >
        Add a factory
      </p>
      <p
        className="FormPromptWithSub__subtitle"
      >
        Include the factory website if it is provided on the product page.
      </p>
    </div>
    <div
      className="FormTextInput"
    >
      <label
        htmlFor="cut-factory-name"
      >
        Factory name
      </label>
      <input
        id="cut-factory-name"
        name="name"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="FormDropdown"
    >
      <label
        htmlFor="cut-factory-location"
      >
        Factory location.  Select 'Not disclosed' if not included on the product page.
      </label>
      
      <select
        name="countryId"
        onChange={[Function]}
        required={true}
        value=""
      />
    </div>
    <div
      className="FormUrlInput"
    >
      <label
        htmlFor="cut-factory-website"
      >
        Factory website.  Leave blank if not included on the product page.
      </label>
      <input
        id="cut-factory-website"
        name="website"
        onChange={[Function]}
        required={true}
        type="url"
        value=""
      />
    </div>
    <div
      className="FormTextarea"
    >
      <label
        htmlFor="cut-factory-notes"
      >
        Are there any additional notes about the cutting factory provided on the product page?
      </label>
      <textarea
        id="cut-factory-notes"
        name="cutNotes"
        onChange={[Function]}
        required={true}
        rows="1"
        value=""
      />
    </div>
    <button
      className="FormButton"
      onClick={[Function]}
      type="submit"
    >
      SUBMIT FACTORY
    </button>
  </section>
  <section
    className="FormPopUp"
    id="cert-pop-up"
  >
    <button
      className="FormPopUp__close-button"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-xmark fa-lg "
        data-icon="xmark"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 384 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
    <header>
      <h1>
        New Certification
      </h1>
    </header>
    <div
      className="FormPromptWithSub"
    >
      <p
        className="FormPromptWithSub__prompt"
      >
        Add a certification
      </p>
      <p
        className="FormPromptWithSub__subtitle"
      >
        
      </p>
    </div>
    <div
      className="FormTextInput"
    >
      <label
        htmlFor="man-cert-name"
      >
        Certification name
      </label>
      <input
        id="man-cert-name"
        name="name"
        onChange={[Function]}
        required={true}
        type="text"
        value={1}
      />
    </div>
    <div
      className="FormUrlInput"
    >
      <label
        htmlFor="man-cert-website"
      >
        Website.  Leave blank if not included on the product page.
      </label>
      <input
        id="man-cert-website"
        name="website"
        onChange={[Function]}
        required={true}
        type="url"
        value=""
      />
    </div>
    <button
      className="FormButton"
      onClick={[Function]}
      type="submit"
    >
      Create certification
    </button>
  </section>
</div>
`;
